WoWsShipBuilder.Common/Features/ShipSelection/FittingToolSelector.razor

```diff
@@ -0,0 +1,176 @@
+                                                                <MudStack Row Spacing="0" Justify="Justify.FlexStart" @key="chunk">
+                                                                    @foreach (var ship in chunk)
+                                                                    {
+                                                                        <div class="ft-selector-ship-name-grid" @key="ship.Index" @onclick="@(_ => SelectShip(ship))">
+                                                                            @if (AppSettings.FittingToolSelectorShowShipImages)
+                                                                            {
+                                                                                <MudImage Src="@($"{CdnOptions.Value.Host}{CdnOptions.Value.ShipImagePath}/{ship.Index}.png")" Height="30" ObjectPosition="ObjectPosition.Center"/>
+                                                                            }
+                                                                            <MudText Class="pa-2" Style="white-space: nowrap; font-size: 1rem">
+                                                                                @Localizer.SimpleGameLocalization(ship.Index)
+                                                                            </MudText>
+                                                                        </div>
+                                                                    }
+                                                                </MudStack>
+                                                            }
+                                                        </MudStack>
+                                                    }
+                                                </MudStack>
+                                                if (tier.Key != shipClass.Value.MaxBy(x => x.Key).Key)
+                                                {
+                                                    <MudDivider Light="false" DividerType="DividerType.FullWidth"/>
+                                                }
+                                            }
+                                        </ul>
+                                    </li>
+                                }
+                            </ul>
+                        </li>
+                    }
+                </ul>
+            </li>
+        }
+    </ul>
+    <MudButton Style="z-index: 900; min-height: 531px; min-width: 20px;" Color="Color.Primary" Variant="Variant.Outlined" Class="px-0" Size="Size.Small" OnClick="@(() => openDrawer = !openDrawer)">
+        @(openDrawer ? "<" : ">")
+    </MudButton>
+</aside>
+
+@code
+{
+    [Parameter]
+    public IList<ShipBuildContainer> SelectedShips { get; set; } = default!;
+
+    private Dictionary<Nation, Dictionary<ShipCategory, Dictionary<ShipClass, Dictionary<int, List<Ship>>>>> treeStructure = new();
```

Comment:
Fix the naming. "treeStructure" does not really help with understanding what it does.
Also, use immutable data structures like ImmutableDictionary